Docs fix metric storage (#13327)
netdata/netdata: database/engine/README.md

> # Database engine

Reviewer: I think it's time for this whole file to come into the database/README.md
I don't see why it need to be a separate file. 
For sure it will make the database/README.md much larger, but I don't see any benefits from having two separate files. dbengine IS our database, the others are side notes for very specific use cases

The Database Engine works like a traditional database. It dedicates a certain amount of RAM for data caching and
indexing, while the rest of the data resides compressed on disk. Unlike other [database modes](/database/README.md), the

> amount of historical metrics stored is based on the amount of disk space you allocate and the effective compression

Reviewer: Too much detail for an opening paragraph. Consider removing it completely. The amount of metrics stored is more complex than this anyway.

ratio, not a fixed number of metrics collected.

> By using both RAM and disk space, the database engine allows for long-term storage of per-second metrics inside of the

Reviewer: I don't like this sentence. consider removing

Agent itself.

In addition, the `dbengine` is the only mode that supports changing the data collection update frequency
(`update every`) without losing the metrics your Agent already gathered and stored.

> ## Tiering

Reviewer 2: Quick glance, let's have the intro just say: 

> Tiering is a mechanism of providing multiple tiers of data with
> different [granularity on metrics](/docs/store/distributed-data-architecture.md#granularity-of-metrics). 
> 
> For Netdata Agents with version `netdata-1.35.0.138.nightly` and greater, `dbengine` supports Tiering, allowing
> almost unlimited retention of data.

Then move the other sentences to the ##Configuration section as the baseline starting piont.

> For Netdata Agent with version `netdata-1.35.0.138.nightly` and greater, `dbengine` supports tiering. Tiering is a

Reviewer: Maybe make this part a bit more marketing oriented (e.g. "unlimited retention")

mechanism of providing multiple tiers of data with
different [granularity on metrics](/docs/store/distributed-data-architecture.md#granularity-of-metrics). Every Tier down

> samples the exact lower tier (lower tiers have greater resolution). You can have up to 5 Tiers **[0. . 4]** of data (

Reviewer: downsamples is one word

Reviewer: There's no "granularity of metrics" link on that document.

Author: it's not in this document, it's in `/docs/store/distributed-data-architecture.md#granularity-of-metrics`, resolving this comment

including the Tier 0, which has the highest resolution) which means years of data.

> ### Metric size

Reviewer: I don't understand the title "Metric size"

Tier 0 is the default that was always available in `dbengine` mode. Tier 1 is the first level of aggregation, Tier 2 is
the second so on so forth.

> Metrics on all tiers > 1 store 5 values for every point for accurate representation:

Reviewer: > 0

hour x 24 hours per day x 7 days per week = 1100MB.

By setting it to `1100`, this node will start maintaining about a week of data. But pay attention to the number of
metrics. If you have more than 2000 metrics on a node, or you need more that a week of high resolution metrics, you may
need to adjust this setting accordingly.

Tier 1 is by default sampling the data every **60 points of Tier 0**. In our case Tier 0 is per second, if we want to
transform this information in terms of time then the Tier 1 "resolution" is per minute.

Tier 1 needs 4 times more storage per point compared to Tier 0. So, for 2000 metrics, with per minute resolution,
retained for a month, Tier 1 needs: 4 bytes x 2000 metrics x 60 minutes per hour x 24 hours per day x 30 days per month
= 330MB.

Tier 2 is by default sampling data every 3600 points of Tier 0 (60 of Tier 1, which is the previous exact Tier). Again
in term of "time" (Tier 0 is per second), then Tier 2 is per hour.

The storage requirements are the same to Tier 1.

For 2000 metrics, with per hour resolution, retained for a year, Tier 2 needs: 4 bytes x 2000 metrics x 24 hours per day
x 365 days per year = 67MB.

> 

Reviewer: We'll probably need the calculator here, not sure. Too many freaking pages!

Author: We present the "tiering mechanism" we don't suggest to the user to implement any change

> ## Configuration

Reviewer: This section looks stupid now. We've already shown the configuration for tiering above. 

Tiering IS the dbengine configuration now, they are not separate things.

Author: Totally agree, we will omit that.

# Database engine

The Database Engine works like a traditional database. It dedicates a certain amount of RAM for data caching and
indexing, while the rest of the data resides compressed on disk. Unlike other [database modes](/database/README.md), the
amount of historical metrics stored is based on the amount of disk space you allocate and the effective compression
ratio, not a fixed number of metrics collected.

## Tiering

For Netdata Agents with version `netdata-1.35.0.138.nightly` and greater, `dbengine` supports Tiering which can offer
almost unlimited retention of data. Tiering is a mechanism of providing multiple tiers of data with
different [granularity on metrics](/docs/store/distributed-data-architecture.md#granularity-of-metrics). Every Tier down
samples the exact lower tier (lower tiers have greater resolution). You can have up to 5 Tiers **[0. . 4]** of data (
including the Tier 0, which has the highest resolution) which means years of data.

### Metric size

Tier 0 is the default that was always available in `dbengine` mode. Tier 1 is the first level of aggregation, Tier 2 is

> the second so on so forth.

Reviewer 2: ```suggestion
the second, and so on.
```

> Metrics on all tiers except of the _Tier 0_, store 5 more values for every point for accurate representation:

Reviewer 2: ```suggestion
Metrics on all tiers except of the _Tier 0_ also store the following five additional values for every point for accurate representation:
```

1. The `sum` of the points aggregated
2. The `min` of the points aggregated
3. The `max` of the points aggregated
4. The `count` of the points aggregated (could be constant, but it may not be due to gaps in data collection)
5. The `anomaly_count` of the points aggregated (how many of the aggregated points found anomalous)

> Among `min`, `max` and `sum`, the right value is chosen based on the user query. `average` is calculated on the fly at

Reviewer 2: ```suggestion
Among `min`, `max` and `sum`, the correct value is chosen based on the user query. `average` is calculated on the fly at
```

query time.

### Tiering in a nutshell

The `dbengine` is capable of retaining metrics for years. To further understand the `dbengine` tiering mechanism let's
explore the following configuration.

```
[db]
    mode = dbengine
    
    # per second data collection
    update every = 1
    
    # enables Tier 1 and Tier 2, Tier 0 is always enabled in dbengine mode
    storage tiers = 3
    
    # Tier 0, per second data for a week
    dbengine multihost disk space MB = 1100
    
    # Tier 1, per minute data for a month
    dbengine tier 1 multihost disk space MB = 330

    # Tier 2, per hour data for a year
    dbengine tier 2 multihost disk space MB = 67
```

For 2000 metrics, collected every second and retained for a week, Tier 0 needs: 1 byte x 2000 metrics x 3600 secs per
hour x 24 hours per day x 7 days per week = 1100MB.

> By setting it to `1100`, this node will start maintaining about a week of data. But pay attention to the number of

Reviewer 2: ```suggestion
By setting `dbengine multihost disk space MB` to `1100`, this node will start maintaining about a week of data. But pay attention to the number of
```

Just making sure we clarify what "it" means.

metrics. If you have more than 2000 metrics on a node, or you need more that a week of high resolution metrics, you may
need to adjust this setting accordingly.

> Tier 1 is by default sampling the data every **60 points of Tier 0**. In our case Tier 0 is per second, if we want to

Reviewer 2: ```suggestion
Tier 1 is by default sampling the data every **60 points of Tier 0**. In our case, Tier 0 is per second, if we want to
```

transform this information in terms of time then the Tier 1 "resolution" is per minute.

> Tier 1 needs 4 times more storage per point compared to Tier 0. So, for 2000 metrics, with per minute resolution,

Reviewer 2: ```suggestion
Tier 1 needs four times more storage per point compared to Tier 0. So, for 2000 metrics, with per minute resolution,
```

    mode = dbengine
```

> To configure the database engine, look for the `page cache size MB` and `dbengine multihost disk space MB` settings in

Reviewer: you reverted it to a wrong setting. The new branch made this `dbengine page cache size MB`

> ### Streaming metrics to the database engine

Reviewer: Since this is the only mode we have now, I don't see much use for this section. 

I'm thinking that maybe we should move together every option that has changed in a "Older versions/Backwards compatibility" section, as they are now in various places.

For more information about setting `[db].mode` on your nodes, in addition to other streaming configurations, see
[streaming](/streaming/README.md).

> ## Requirements & limitations

Reviewer: I think that almost all of this needs to go away, or be left in a "Legacy/Older versions" section/page. Have a chat with Costa or Stelio about this first though.

Author: We are getting out of scope, 

I will open a new issue on the dbengine limitations. People also asking the limitation/specs on _how many clients a parent can handle_ which is correlated to software requirements/limitations and resources dbengine needs

The DB engine stores chart metric values in 4096-byte pages in memory. Each chart dimension gets its own page to store
consecutive values generated from the data collectors. Those pages comprise the **Page Cache**.

> When those pages fill up they are slowly compressed and flushed to disk. It can

Reviewer 2: ```suggestion
When those pages fill up, they are slowly compressed and flushed to disk. It can
```
